src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -117,26 +117,41 @@
     Optional<StudentParticipation> findByExerciseIdAndTeamIdWithLatestResult(@Param("exerciseId") Long exerciseId, @Param("teamId") Long teamId);
 
     /**
+     * todo SE, NR: maybe re-add
+     * 'and not exists (select prs from participation.results prs where prs.assessmentType IN ('MANUAL', 'SEMI_AUTOMATIC'))'
+     *
      * Find all participations of submissions that are submitted and do not already have a manual result and do not belong to test runs.
      * No manual result means that no user has started an assessment for the corresponding submission yet.
      *
      * If a student can have multiple submissions per exercise type, the latest submission (by id) will be returned.
      *
+     * @param correctionRound the correction round the fetched results should belong to
      * @param exerciseId the exercise id the participations should belong to
      * @return a list of participations including their submitted submissions that do not have a manual result
      */
     @Query("""
             select distinct participation from StudentParticipation participation
             left join fetch participation.submissions submission
-            left join fetch submission.results result
-            left join fetch result.feedbacks feedbacks
+            left join fetch submission.results results
+            left join fetch results.feedbacks feedbacks
             where participation.exercise.id = :#{#exerciseId}
+            and 0L = (SELECT COUNT(r2)
+                             FROM Result r2 where r2.assessor IS NOT NULL
+                                 AND (r2.rated IS NULL OR r2.rated = FALSE)
+                                 AND r2.submission = submission)
+            and
+              :#{#correctionRound} = (SELECT COUNT(r)
+                             FROM Result r where r.assessor IS NOT NULL
+                                 AND r.rated = TRUE
+                                 AND r.submission = submission
+                                 AND r.completionDate IS NOT NULL AND
+                                 (participation.exercise.dueDate IS NULL OR r.submission.submissionDate <= participation.exercise.dueDate))
             and not exists (select prs from participation.results prs where prs.assessor.id = participation.student.id)
-            and not exists (select prs from participation.results prs where prs.assessmentType IN ('MANUAL', 'SEMI_AUTOMATIC'))
```

Comment:
How to you ensure that only assessments which do not have a manual result are selected if you remove this? Was this redundant if you check for the assessor?

Reply:
We added it inside the subquery now